### ISSUE TITLE
[FR] Bibtex citations support

### ISSUE THREAD

**Issue by bambirombi** [artempyanykh/marksman#189]
Something like [pandocciter](https://marketplace.visualstudio.com/items?itemName=notZaki.pandocciter) would be amazing.

Citation syntax: @BibKey or [@bibKey]

**Comment by artempyanykh** [comment]
@bambirombi 👍👍 I was thinking about this for a while

**Comment by jjakubassa** [comment]
I agree, the autocompletion would be awesome. Perhaps this could be complemented with a autogenerated bibliography, similar to the toc generation and similar to [this project](https://github.com/renerocksai/sublimeless_zk?tab=readme-ov-file#automatic-bibliographies). 

The conversion of cite key to full citation could be done via pandoc. 

<img width="1312" alt="image" src="https://github.com/user-attachments/assets/d94f35a5-7df6-448b-b29c-a91e653fd7e3">
